### PR TITLE
Add `syncUsers` field to pgbouncer api

### PR DESCRIPTION
**Pull request by HiranmoyChowdhury**
Signed-off-by: Hiranmoy Das Chowdhury <hiranmoy@appscode.com>

### CODE REVIEW

File: apis/kubedb/v1alpha2/pgbouncer_types.go [hunk 1]
```diff
@@ -118,10 +118,17 @@ const (
 )
 
 type Databases struct {
+	// SyncUsers is a boolean type and when enabled, operator fetches all users created in the backend server to the
+	// PgBouncer server. Password changes are also synced in pgBouncer when it is enabled.
```
Comment (MobarakHsn) [inline apis/kubedb/v1alpha2/pgbouncer_types.go:122]: fetches Postgres users from externally managed secrets and sync them to PgBouncer.

**Review by MobarakHsn — CHANGES_REQUESTED** [review]

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]
Run `go mod tidy && go mod vendor` , & `make gen fmt`

File: apis/kubedb/v1alpha2/pgbouncer_types.go (hunk 1 above)
Comment (HiranmoyChowdhury) [reply apis/kubedb/v1alpha2/pgbouncer_types.go]: @MobarakHsn I got this comment from pgpool though. I already replaced the comment. Can you review it again if it's ok?

**Review by ArnobKumarSaha — APPROVED** [review]

File: apis/kubedb/v1alpha2/pgbouncer_types.go (hunk 1 above)
Comment (MobarakHsn) [reply apis/kubedb/v1alpha2/pgbouncer_types.go]: It was previous mechanism for Pgpool now it works differently.

**Pull request merged**